src/components/main/note/EditableTextInput.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { forwardRef, useImperativeHandle, useRef } from 'react';
+import ReactQuill from 'react-quill';
+import 'react-quill/dist/quill.bubble.css';
+
+import DoneIcon from '@mui/icons-material/Done';
+import { Container } from '@mui/material';
+import IconButton from '@mui/material/IconButton';
+import Stack from '@mui/material/Stack';
+
+import './note_style.css';
+
+interface EditableTextInputProps {
+  value: string;
+  onChange: (t: string) => void;
+  onKeyDown: (e: React.KeyboardEvent) => void;
+}
+
+export type EditableTextInputRefType = {
+  focus: () => void;
+};
+
+const EditableTextInput = forwardRef<
+  EditableTextInputRefType,
+  EditableTextInputProps
+>((props: EditableTextInputProps, ref) => {
+  const { value, onChange, onKeyDown } = props;
+  const textInput = useRef<ReactQuill>(null);
+
+  const modules = {
+    toolbar: [
+      ['bold', 'italic', 'underline'],
+      [{ list: 'ordered' }, { list: 'bullet' }],
+      ['link'],
+      ['clean'],
+    ],
+  };
+
+  const formats = [
+    'header',
+    'bold',
+    'italic',
+    'underline',
+    'list',
+    'bullet',
+    'link',
+  ];
+
+  const focus = (): void => {
+    textInput.current?.focus();
+  };
+
+  useImperativeHandle(ref, () => ({
+    focus,
+  }));
+
+  return (
+    <Stack direction="row" justifyContent="space-between">
+      <ReactQuill
+        theme="bubble"
+        value={value}
+        onChange={onChange}
+        onKeyDown={onKeyDown}
+        ref={textInput}
+        modules={modules}
+        formats={formats}
+      />
+      <Container>
+        <IconButton>
+          <DoneIcon />
+        </IconButton>
+      </Container>
+    </Stack>
+  );
+});
+
+EditableTextInput.displayName = 'EditableTextInput';
```

Comment:
This defines the name in the react inspector right? But here it's exactly the same name as the component?

Reply:
As mentionned [here](https://github.com/graasp/graasp-app-sticky-notes/pull/78#discussion_r1023679752), I think that it is just for react to be able to determine a name for the component despite the reference forwarding.